lib/msf/core/handler/bind_aws_instance_connect.rb

```diff
@@ -0,0 +1,347 @@
+  end
+
+  #
+  # Returns the connection oriented general handler type, in this case bind.
+  #
+  def self.general_handler_type
+    "bind"
+  end
+
+  # A string suitable for displaying to the user
+  #
+  # @return [String]
+  def human_name
+    "Bind AWS InstanceConnect"
+  end
+
+  #
+  # Initializes a bind handler and adds the options common to all bind
+  # payloads, such as local port.
+  #
+  def initialize(info = {})
+    super
+
+    register_options(
+      [
+        OptString.new('EC2_ID', [true, 'The EC2 ID of the instance ', '']),
+        OptString.new('REGION', [true, 'AWS region containing the instance', 'us-east-1']),
+        OptString.new('ACCESS_KEY_ID', [false, 'AWS access key', nil]),
+        OptString.new('SECRET_ACCESS_KEY', [false, 'AWS secret key', nil]),
+        OptString.new('INSTANCE_USER', [false, 'Username on the EC2 instance with which to log-in']),
+        OptString.new('INSTANCE_PORT', [false, 'Serial port index on the EC2 instance to use for the connection']),
+        OptString.new('ROLE_ARN', [false, 'AWS assumed role ARN', nil]),
+        OptString.new('ROLE_SID', [false, 'AWS assumed role session ID', nil]),
+      ], Msf::Handler::BindAwsInstanceConnect)
+
+    register_advanced_options(
+      [
+        OptString.new('PRIVATE_KEY', [
+          false,
+          'The string value of the private key that will be used. If you are using MSFConsole,
+          this value should be set as file:PRIVATE_KEY_PATH. OpenSSH, RSA, DSA, and ECDSA private keys are supported.'
+        ]),
+        OptString.new('KEY_PASS', [false, 'Passphrase for SSH private key(s)']),
+        OptBool.new('SSH_DEBUG', [ false, 'Enable SSH debugging output (Extreme verbosity!)', false])
+      ], Msf::Handler::BindAwsInstanceConnect)
+
+    self.listener_threads = []
+    self.conn_threads = []
+    self.listener_pairs   = {}
+  end
+
+  #
+  # Kills off the connection threads if there are any hanging around.
+  #
+  def cleanup_handler
+    # Kill any remaining handle_connection threads that might
+    # be hanging around
+    stop_handler
+    conn_threads.each { |thr|
+      thr.kill
+    }
+  end
+
+  #
+  # Starts a new connecting thread
+  #
+  def add_handler(opts={})
+
+    # Merge the updated datastore values
+    opts.each_pair do |k,v|
+      datastore[k] = v
+    end
+
+    # Start a new handler
+    start_handler
+  end
+
+  #
+  # Starts monitoring for an outbound connection to become established.
+  #
+  def start_handler
+
+    # Maximum number of seconds to run the handler
+    ctimeout = 150
+
+    # Maximum number of seconds to await initial API response
+    rtimeout = 5
+
+    if (exploit_config and exploit_config['active_timeout'])
+      ctimeout = exploit_config['active_timeout'].to_i
+    end
+    self.listener_pairs[datastore['EC2_ID']] = true
+
+    # Start a new handling thread
+    self.listener_threads = framework.threads.spawn("BindAwsInstanceConnectHandler-#{datastore['EC2_ID']}", false) {
```

Comment:
It looks like `framework.threads.spawn(...)` would return a single thread, is there a naming conflict here? i.e. should this be `listener_thread`


Or should this be `self.listener_threads << framework.threads.spawn(...)`

Reply:
From the last time I looked at this it's meant to be this:

```suggestion
    self.listener_threads << framework.threads.spawn("BindAwsInstanceConnectHandler-#{datastore['EC2_ID']}", false) {
```

Potentially a copy/pasta mix-up

Reply:
Thanks - fixed